lib/Dialect/RTL/Dialect.cpp

```diff
@@ -15,6 +15,25 @@ using namespace rtl;
 // Dialect specification.
 //===----------------------------------------------------------------------===//
 
+namespace {
+
+// We implement the OpAsmDialectInterface so that RTL dialect operations
+// automatically interpret the name attribute on operations as their SSA name.
+struct RTLOpAsmDialectInterface : public OpAsmDialectInterface {
+  using OpAsmDialectInterface::OpAsmDialectInterface;
+
+  /// Get a special name to use when printing the given operation. See
+  /// OpAsmInterface.td#getAsmResultNames for usage details and documentation.
+  void getAsmResultNames(Operation *op,
+                         OpAsmSetValueNameFn setNameFn) const override {
+    // If an operation have an optional 'name' attribute, use it.
+    if (isa<WireOp>(op) && op->getNumResults() > 0)
```

Comment:
Random nit, but getNumResults() is always 1 for wires, so this extra check isn't needed.

Reply:
Sorry, I fixed in #147